core/cache/src/main/java/uk/gov/gchq/gaffer/cache/Cache.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.cache;
+
+import uk.gov.gchq.gaffer.cache.exception.CacheOperationException;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Type safe cache, adding and getting is guaranteed to be same type.
+ * @param <V> The type of values to add and get.
+ */
+public class Cache<V> {
+    public static final String ERROR_ADDING_KEY_TO_CACHE_KEY_S = "Error adding key to cache. key: %s";
+    protected String cacheServiceName;
+
+    public Cache(final String cacheServiceName) {
+        this.cacheServiceName = cacheServiceName;
```

Comment:
Should this variable be cacheName rather than cacheServiceName?

Reply:
name carried over from commit ddabdf72 FederatedStoreCache.CACHE_SERVICE_NAME
Would you like me to change it.